compiler/ETA/Main/DriverPipeline.hs

```diff
@@ -1293,6 +1296,7 @@ linkGeneric dflags oFiles depPackages = do
     --        (text $ "    Call hsInit() from your main() method to set"
     --          ++ " these options."))
     -- TODO: Use conduits to combine the jars
+    mapM_ warnHscFile $ filter (".hsc" `isSuffixOf`) oFiles
```

Comment:
This should be extended to fail (via throwing an exception) or use if-else and make sure nothing else happens if an `.hsc` file is detected.